src/common/MessageProcessor.sol

```diff
@@ -68,11 +68,9 @@ contract MessageProcessor is Auth, IMessageProcessor {
     /// @inheritdoc IMessageHandler
     function handle(uint16 centrifugeId, bytes calldata message) external auth {
         MessageType kind = message.messageType();
+        uint16 sourceCentrifugeId = message.messageSourceCentrifugeId();
 
-        if (message.messageDirection() == MessageDirection.HubToVaults) {
-            // Hub => Vaults messages should only be able to interact with pools from that Hub chain
-            require(message.messagePoolId().centrifugeId() == centrifugeId, InvalidSourceChain());
-        }
+        require(sourceCentrifugeId == 0 || sourceCentrifugeId == centrifugeId, InvalidSourceChain());
```

Comment:
It means: if there is some centrifugeId attached to the message as source one, then check against the adapter one